From bbbd777fd19f19e90a885aea8d9c928b20e411ca Mon Sep 17 00:00:00 2001
From: Matej Habrnal <mhabrnal@redhat.com>
Date: Tue, 16 Jan 2018 14:02:09 +0100
Subject: [PATCH] problem_data: fix segfault if last_occurrence doesn't exist

If creating vmcore problem by the following scenario:
https://tcms.engineering.redhat.com/case/315493/?from_plan=21410
last_occurrence element doesn't exist. This causes abrt-cli list
segfault. This commit fixes this issue.

Resolves #1421754

Signed-off-by: Matej Habrnal <mhabrnal@redhat.com>
---
 src/lib/problem_data.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/lib/problem_data.c b/src/lib/problem_data.c
index 8aa2ed9..e7a4a8b 100644
--- a/src/lib/problem_data.c
+++ b/src/lib/problem_data.c
@@ -478,11 +478,11 @@ gint cmp_problem_data(gconstpointer a, gconstpointer b, gpointer filename)
 {
     problem_data_t *a_data = *(problem_data_t **) a;
     const char *a_time_str = get_problem_item_content_or_NULL(a_data, filename);
-    unsigned long a_time= strtoul(a_time_str, NULL, 10);
+    unsigned long a_time = a_time_str ? strtoul(a_time_str, NULL, 10) : 0;
 
     problem_data_t *b_data = *(problem_data_t **) b;
     const char *b_time_str = get_problem_item_content_or_NULL(b_data, filename);
-    unsigned long b_time= strtoul(b_time_str, NULL, 10);
+    unsigned long b_time = b_time_str ? strtoul(b_time_str, NULL, 10) : 0;
 
     /* newer first */
     if (a_time > b_time)
-- 
2.14.3

